 linker/linker.cpp | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/linker/linker.cpp b/linker/linker.cpp
index 99c02d8..2f3d0a1 100644
--- a/linker/linker.cpp
+++ b/linker/linker.cpp
@@ -3014,12 +3014,6 @@ bool soinfo::link_image(const soinfo_list_t& global_group, const soinfo_list_t&
     // Fail if app is targeting sdk version > 22
     // TODO (dimitry): remove != __ANDROID_API__ check once http://b/20020312 is fixed
 #if !defined(__i386__) // ffmpeg says that they require text relocations on x86
-    if (get_application_target_sdk_version() != __ANDROID_API__
-        && get_application_target_sdk_version() > 22) {
-      PRINT("%s: has text relocations", get_realpath());
-      DL_ERR("%s: has text relocations", get_realpath());
-      return false;
-    }
 #endif
     // Make segments writable to allow text relocations to work properly. We will later call
     // phdr_table_protect_segments() after all of them are applied and all constructors are run.
-- 
2.1.4

